src/test/kotlin/org/veupathdb/vdi/test/ApiTest.kt

```diff
@@ -0,0 +1,158 @@
+package org.veupathdb.vdi.test
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import io.restassured.RestAssured
+import io.restassured.RestAssured.given
+import io.restassured.http.ContentType
+import org.apache.logging.log4j.kotlin.logger
+import org.awaitility.Awaitility.await
+import org.awaitility.core.ConditionTimeoutException
+import org.junit.jupiter.api.AfterAll
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.TestInstance
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.MethodSource
+import java.net.URL
+import java.nio.file.FileVisitOption
+import java.nio.file.Files
+import java.nio.file.Path
+import java.time.Duration
+import java.util.stream.Stream
+import kotlin.io.path.isDirectory
+import kotlin.io.path.name
+
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class ApiTest {
+    private val AuthToken: String = System.getProperty("AUTH_TOKEN")
+    private val AuthTokenKey: String = "Auth-Key"
+    private val ObjectMapper = ObjectMapper()
+
+    // Files in the testdata resource directory must follow structure testdata/{dataset-type}/{upload-file}
+    private val TestFilesDir = "testdata"
+
+    // Keep track of all datasets created to clean them after all tests are done.
+    private val DatasetsToClean = mutableListOf<String>()
+
+    @BeforeAll
+    internal fun setup() {
+        RestAssured.baseURI = System.getProperty("BASE_URL")
+    }
+
+    @AfterAll
+    internal fun teardown() {
+        DatasetsToClean.forEach {
+            logger().info("Cleaning dataset $it")
+            given()
+                .header("Auth-Key", System.getProperty("AUTH_TOKEN"))
+                .`when`()
+                .delete("vdi-datasets/$it")
+                .then()
+                .statusCode(204)
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("fileProvider")
+    fun parameterizedTest(input: TestCase) {
+        val meta: Map<String, Any> = mapOf(
+            Pair(
```

Comment:
FYI: There is a `to` infix method for constructing `Pair` instances that is generally the standard for use cases like this.  Not in anyway required, just mentioning it in case you weren't already aware.

Example:
```kotlin
val foo = mapOf(
  "foo"     to "bar",
  "fizz"    to "buzz",
  "animals" to listOf("cat", "rat", "dog"),
)
```